Source/Evolve/WarpXEvolve.cpp

```diff
@@ -262,6 +238,29 @@ WarpX::Evolve (int numsteps)
             ExecutePythonCallback("afterEsolve");
         }
 
+        if (evolve_scheme == EvolveScheme::Explicit) {
+            if (cur_time + dt[0] >= stop_time - 1.e-3*dt[0] || step == numsteps_max-1) {
```

Comment:
```suggestion
            if (cur_time >= stop_time - 1.e-3*dt[0] || step == numsteps_max-1) {
```
Since `cur_time` is already updated.